client/lib/signup/step-actions/index.js

```diff
@@ -388,6 +388,10 @@ export function createAccount(
 	},
 	reduxStore
 ) {
+	if ( flowName === 'p2' ) {
+		flowName = 'wp-for-teams';
```

Comment:
Ok so I'm not sure this is the best way as it could introduce some hidden technical debt (ie people might assume the flow name is obtained from the flow name in `client/signup/config/flows-pure.js` but for P2 it's modified here).

On the other hand, we rely on the value to be `wp-for-teams` on too many places like:
- all the marketing emails (lots of places on wpcom codebase and especially Iterable)
- parts of Calypso
- the P2020 theme itself

What do you think — should we take the effort (and pings!) to update the value everywhere now or do this?

Reply:
Would it work and would the code be slightly more grokkable to people if we just add the `p2` flow (alongside the existing `wp-for-teams` flow) inside `flows-pure.js`?  Maybe something like `flows['p2'] = { ...flows['wp-for-teams'] }` after `flows['wp-for-teams']` is defined?

But yeah, I'm starting to worry about having to refactor all the "wpforteams" tokens we used during early development. Also about some issues like if we go for consistency and change `isWPForTeams` to `isP2` would it lead some to think it refers to the theme and not the product?

Reply:
> Would it work and would the code be slightly more grokkable to people if we just add the p2 flow (alongside the existing wp-for-teams flow) inside flows-pure.js? Maybe something like flows['p2'] = { ...flows['wp-for-teams'] } after flows['wp-for-teams'] is defined?

We could, but if you navigate to `start/p2` and go through that signup flow, the `signup_flow_name` will be set to `p2`, not `wp-for-teams` (what we need) without the above change.

Or do you mean to add that flow definition in addition to making the above change? Yeah, that might be a bit confusing and clarifying at the same time 😅 . Like "why do we need two flows for the same steps?". But if we add comment explanation, that could help.

> ut yeah, I'm starting to worry about having to refactor all the "wpforteams" tokens we used during early development. Also about some issues like if we go for consistency and change isWPForTeams to isP2 would it lead some to think it refers to the theme and not the product?

I think we should not worry about refactoring all the functions and places where we have "wpforteams" (like `isWPForTeamsSite`) as that's a code "artifact". When we built Atomic, originally, it was called Automated Transfer and you'll find that name scattered all across the codebase.

The reason I brought this up is that the `wp-for-teams` signup flow name works like a constant in many places (especially for Marketing) so if you change that, we have to update all the backend places and ping Marketing to update it in all the Iterable places too.

Reply:
> change isWPForTeams to isP2 would it lead some to think it refers to the theme and not the product?

Yeah, good point. I'm still using "wp for teams" everywhere where I need to check for a site or some other core logic. I think it's good for consistency's sake.